packages/algorand-js/src/interpreter/interpreter.ts

```diff
@@ -105,20 +50,18 @@ export class Interpreter {
   }
 
   /**
-   * Description: this function executes set of Operator[] passed after
-   * parsing teal code
-   * @param {ExecParams} txn : Transaction parameters
-   * @param {Logic[]} logic : smart contract instructions
-   * @param {AppArgs} args : external arguments
-   * @returns {boolean} : transaction accepted/rejected based on ASC logic
+   * Description: this function executes TEAL code after parsing
+   * @param {string} program: teal code
+   * @param {Uint8Array[]} args : external arguments
+   * @param {Runtime} runtime : runtime object
+   * @param {Context} context: interpreter context (state and txns)
    */
-  execute (txnParams: ExecParams | ExecParams[],
-    logic: Operator[], args: Uint8Array[],
-    accounts: AccountState[]): boolean {
+  async execute (program: string, args: Uint8Array[],
```

Comment:
I think `args` should be put in context also.

Reply:
`args` are provided by the user. We are keeping the state we prepared in the `ctx` (eg, `encoded tx, gtxs`, `global assets, apps..`). Is it okay to put `args` inside ctx?

Reply:
I think so, because args are also runtime context, specific to each transaction.

Reply:
Ok, moving `args` to ctx.